aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -81,14 +86,38 @@ public S3FileIO(SerializableSupplier<S3Client> s3, AwsProperties awsProperties)
     this.awsProperties = awsProperties;
   }
 
+  /**
+   * Constructor with custom s3 supplier and s3 tags
+   * <p>
+   * Calling {@link S3FileIO#initialize(Map)} will overwrite information set in this constructor.
+   * param s3 s3 supplier
+   * @param properties catalog properties
+   */
+  public S3FileIO(SerializableSupplier<S3Client> s3, Map<String, String> properties) {
```

Comment:
the logic should be moved to `initialize`

Reply:
Thanks, I have made the changes.